### PR TITLE
Incorrect bounding box calculations for very small cylindrical pixels

### LINKED ISSUE #27552: Incorrect bounding box calculations for very small cylindrical pixels
For IDF containing detector pixels in the shape of cylinders of diameter 5.5 mm and height 4.2, it was found that:
```
shape = workspace.componentInfo().shape(42)  # shape object for pixel detector 42
shape.getBoundingBox().width()
```
results in
[0.0055,0.00420189,0.0055]
The height of the pixel is off by extraneous '0.189'. Although not much, it prevents accurate calibration of pixel widths with the bar scans method.

### PR DESCRIPTION
**Pull request by jmborr**
**Description of work.**
- [x] new expressions for line intersection with plane
- [x] unit test for bounding box calculation for a capped cylinder detector
- [x] amended [release notes](https://github.com/mantidproject/mantid/blob/9126b057c92cc120344250d81995860c7e2d015b/docs/source/release/v4.3.0/framework.rst#id3)

**To test:**
Upload and unzip the [test IDF](https://github.com/mantidproject/mantid/files/3934612/test_instrument.xml.zip), then run the following script in MantidPlot or the Workbench:
```
w = LoadEmptyInstrument(Filename='/tmp/test_instrument.xml', OutputWorkspace='test')
print(w.componentInfo().shape(42).getBoundingBox().width())
```
Results should be `[0.0055, 0.0042, 0.0055]`

Fixes #27552

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by jmborr** [comment]
@AndreiSavici would you please take a look to this PR?

File: Framework/Algorithms/test/ReflectometryReductionOne2Test.h [hunk 1]
```diff
@@ -465,7 +465,7 @@ class ReflectometryReductionOne2Test : public CxxTest::TestSuite {
     alg.setProperty("ThetaIn", 25.0);
     MatrixWorkspace_sptr outLam = runAlgorithmLam(alg, 12);
 
-    TS_ASSERT_DELTA(outLam->x(0)[0], 0.934991, 1e-6);
+    TS_ASSERT_DELTA(outLam->x(0)[0], 0.934992, 1e-6);
```
Comment (peterfpeterson) [inline Framework/Algorithms/test/ReflectometryReductionOne2Test.h]: I'm surprised you had to update the expected values at the level of tolerance.

File: Framework/Geometry/src/Surfaces/Plane.cpp [hunk 2]
```diff
@@ -306,14 +306,24 @@ int Plane::LineIntersectionWithPlane(V3D startpt, V3D endpt, V3D &output) {
   double sprod = this->getNormal().scalar_prod(startpt - endpt);
   if (sprod == 0)
     return 0;
-  double s1 = (NormV[0] * startpt[0] + NormV[1] * startpt[1] +
-               NormV[2] * startpt[2] - Dist) /
-              sprod;
+  double projection =
```
Comment (peterfpeterson) [inline Framework/Geometry/src/Surfaces/Plane.cpp]: Thank you for improving names and breaking up the calculation. You could `const` these to allow the compiler to do its magic.

File: Framework/Geometry/src/Surfaces/Plane.cpp [hunk 3]
```diff
@@ -306,14 +306,24 @@ int Plane::LineIntersectionWithPlane(V3D startpt, V3D endpt, V3D &output) {
...
+      NormV[0] * startpt[0] + NormV[1] * startpt[1] + NormV[2] * startpt[2];
+  double s1 = (projection - Dist) / sprod;
   if (s1 < 0 || s1 > 1)
     return 0;
+  /*Action: deprecated expressions below. Reason: incur in numerical errors when
```
Comment (peterfpeterson) [inline Framework/Geometry/src/Surfaces/Plane.cpp]: Remove the old code and add a comment about why the new way is numerically correct

File: Framework/Geometry/test/ComponentInfoTest.h [hunk 4]
```diff
@@ -332,6 +332,53 @@ class ComponentInfoTest : public CxxTest::TestSuite {
     TS_ASSERT(boundingBox.isNull());
   }
 
+  // Test calculation of the bounding box for a milimiter-sized
```
Comment (peterfpeterson) [inline Framework/Geometry/test/ComponentInfoTest.h]: :+1:

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Very minor changes requested. Is ready to merge otherwise.

File: Framework/Algorithms/test/ReflectometryReductionOne2Test.h (hunk 1 above)
Comment (jmborr) [reply Framework/Algorithms/test/ReflectometryReductionOne2Test.h]: yes, but it's very minor. One would have to ask to the guy who wrote this tests if 1e-6 was really a physically significant tolerance

File: Framework/Geometry/src/Surfaces/Plane.cpp (hunk 2 above)
Comment (jmborr) [reply Framework/Geometry/src/Surfaces/Plane.cpp]: done

File: Framework/Geometry/src/Surfaces/Plane.cpp (hunk 3 above)
Comment (jmborr) [reply Framework/Geometry/src/Surfaces/Plane.cpp]: done

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**